morpheus/models/dfencoder/autoencoder.py

```diff
@@ -737,251 +741,132 @@ def _create_stat_dict(self, a):
         scaler.fit(a)
         return {'scaler': scaler}
 
-    def fit(
+    def _transform_dataset_for_training(self, dataset):
+        dataset.batch_size = self.batch_size
+        dataset.preprocess_fn = self.preprocess_train_data
+        dataset.shuffle_batch_indices = True
+        dataset.shuffle_rows_in_batch = True
+
+        return dataset
+
+    def _transform_dataset_for_validation(self, dataset):
+        dataset.batch_size = self.eval_batch_size
+        dataset.preprocess_fn = self.preprocess_validation_data
+        dataset.shuffle_batch_indices = False
+        dataset.shuffle_rows_in_batch = False
+
+        return dataset
```

Comment:
Ok I see why you need these setters on the dataset classes. I still feel that it would be better to set these in the constructor since they are used in `__iter__`. If you changed these properties between iterations, you could introduce bugs

Reply:
I agree, but anyone changing things between iterations is making a conscious decision to do so, I don't necessarily think we should avoid these transforms as they save us copying the dataset otherwise.

Reply:
Fair enough